sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/CompressibleColumnBuilder.scala

```diff
@@ -46,6 +47,7 @@ private[columnar] trait CompressibleColumnBuilder[T <: AtomicType]
   this: NativeColumnBuilder[T] with WithCompressionSchemes =>
 
   var compressionEncoders: Seq[Encoder[T]] = _
+  private val alignedArch = Set("sparc", "sparcv9").contains(System.getProperty("os.arch"))
```

Comment:
Not sure what you mean. In https://github.com/apache/spark/pull/14762#r77049945 I suggested the change you made here, and that it be in a companion object. But here it's in the class. I'm talking about companion objects as in "static methods". You don't need this same flag value in every instance of the class.